test/unit/module/core/simd_cast.cpp

```diff
@@ -117,30 +129,71 @@ simd_cast_logical_test()
   simd_cast_test<eve::logical<T>, eve::logical<U>>();
 }
 
+template<typename Wide>
+consteval bool
+rvv_check_enable()
+{
+  using scalar_type = typename Wide::value_type;
+  return (eve::fundamental_cardinal_v<scalar_type> == eve::cardinal_v<Wide>
+          || eve::expected_cardinal_v<scalar_type> == eve::cardinal_v<Wide>);
+}
```

Comment:
This check is way to restrictive. 

I don't think the suggested in this pull request approach to deal with rvv testing being problematic, is a good approach. Yeah we need to solve it somehow - but you disable like almost everything.

Reply:
Any logical can be simd cast to any other logical. 
We shouldn't remove the tests here. 

If there is too many cases, we should just globally test less.
I think I mentioned how I suggested to reduce the global test cases somewhere.

Reply:
> Any logical can be simd cast to any other logical

I'm checking logical's `simd_cast` availability in `simd_cast_logical_test()`, so for `logical` and` wide` the same combinations of types are tested.

> I think I mentioned how I suggested to reduce the global test cases somewhere.

All the tests you have suggested are enabled (check `test_enable` template function - it enables tests for `wide`):

> all tests below 256 bytes

` if constexpr( sizeof(T) <= size_limit && sizeof(U) <= size_limit ) return true;`

> 16 doubles, 32 doubles, 16 uint64s, 32 uint64s

`test_enable_cardinals` enables testing for all types with expected cardinal (including 16 doubles, 16 uint64s).
Furthermore, testing is enabled for 32 doubles, 32 int64, 32 int64 (that have `cardinal` == `expected_cardinal` * 2).

Reply:
You keep not hearing me.

simd_cast is very small test.
If we are having troubles going through all of the cases here and you have to do things like:

```
    constexpr auto size_limit = 256;
    if constexpr( sizeof(T) <= size_limit && sizeof(U) <= size_limit ) return true;
    else return test_enable_cardinals<T>();
```

then we are going to have troubles in every test.

I means we need to reduce the overall number of tests drammaticaly. 

So - we should reduce the amount of test cases requested for rvv. 
The `simd::all_types` and the other one too should change.

I suggested: 

all tests below 256 bytes.
16 doubles, 32 doubles, 16 uint64s, 32 uint64s
The rest is left untested.

Reply:
Oh, yep, I thought to change `simd_cast` only. I will update review.
In this context, I have a question:
32 uint64s, 32 doubles - these types are not the part of `eve::test::simd::all_types`, as they are aggregated (for riscv rvv with vlen=128). You mean that I need to modify test that check aggregated, or better to modify `simd::all_types` to have `expected_cardinal` and `expected_cardinal/2`?

Reply:
Thye should include aggregated. 
Some modification here is needed for rvv:
https://github.com/jfalcou/eve/blob/031592288699acb36cb833fa23510eb028765dcf/test/test.hpp#L176

@jfalcou please  help

Reply:
expected_cardinals for rvv is enourmous, I don't think testing it, except for int64 and double is something we should do. Just too slow.